app/models/comment.rb

```diff
@@ -84,12 +85,24 @@ def no_grandparents
     errors.add(:parent, 'cannot have a parent of their own') if parent&.parent
   end
 
+  def ama_closed
+    return unless post
+    return unless post.ama
+    ama = post.ama
+    return if ama.author == user
+    now_time = Time.now
+
+    unless ama.start_date <= now_time && ama.end_date >= now_time
+      errors.add(:post, 'cannot make any more comments on this ama')
+    end
+  end
+
```

Comment:
Perhaps it would make more sense to model this whole thing outside of the Post/Comment structure (since you have the AMA model already), so that it can have more logic specifically geared towards that?  At the very least, this should go on the AMA, so you can just ask `ama.closed?`